src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentsService.kt

```diff
@@ -158,20 +173,54 @@ class AppointmentsService(
     }
   }
 
-  private fun getActionPlanAppointmentOrThrowException(actionPlanId: UUID, sessionNumber: Int): ActionPlanAppointment =
+  private fun getActionPlanSessionOrThrowException(actionPlanId: UUID, sessionNumber: Int): ActionPlanSession =
 
     getActionPlanAppointment(actionPlanId, sessionNumber)
       ?: throw EntityNotFoundException("Action plan appointment not found [id=$actionPlanId, sessionNumber=$sessionNumber]")
 
   private fun getActionPlanAppointment(actionPlanId: UUID, sessionNumber: Int) =
-    actionPlanAppointmentRepository.findByActionPlanIdAndSessionNumber(actionPlanId, sessionNumber)
+    actionPlanSessionRepository.findByActionPlanIdAndSessionNumber(actionPlanId, sessionNumber)
 
   private fun mergeAppointment(
-    appointment: ActionPlanAppointment,
+    actionPlanSession: ActionPlanSession,
     appointmentTime: OffsetDateTime?,
     durationInMinutes: Int?
   ) {
-    appointmentTime?.let { appointment.appointmentTime = it }
-    durationInMinutes?.let { appointment.durationInMinutes = it }
+    appointmentTime?.let { actionPlanSession.appointment.appointmentTime = it }
+    durationInMinutes?.let { actionPlanSession.appointment.durationInMinutes = it }
+  }
+
+  fun createInitialAssessment(
+    referral: Referral,
+    createdByUser: AuthUser
+  ): Referral {
+
+    referral.supplierAssessment = supplierAssessmentRepository.save(
+      SupplierAssessment(
+        id = UUID.randomUUID(),
+        referral = referral,
+        appointments = setOf(
+          appointmentRepository.save(
+            Appointment(
+              id = UUID.randomUUID(),
+              createdBy = authUserRepository.save(createdByUser),
+              createdAt = OffsetDateTime.now()
+            )
+          )
+        ),
+      )
+    )
+    return referralRepository.save(referral)
+  }
+
+  fun updateInitialAssessment(
+    referral: Referral,
+    durationInMinutes: Int?,
+    appointmentTime: OffsetDateTime?
+  ): SupplierAssessment {
+    durationInMinutes?.let { referral.supplierAssessment!!.appointment.durationInMinutes = durationInMinutes }
```

Comment:
Not sure if using '!!' here is the correct way to handle this. There should always be a supplier assessment at this point due to it being created upon sending a referral